docs/notebooks/Calculating ground states on large scale systems with the quantum krylov subspace method.ipynb

```diff
@@ -0,0 +1,1037 @@
+    "\n",
+    "$$\\langle \\psi_m \\vert \\hat{H} \\vert \\psi_n \\rangle$$\n",
+    "$$\\langle \\psi \\vert  e^{i \\hat{H} t_m}   \\hat{H} e^{-i \\hat{H} t_n} \\vert \\psi \\rangle$$\n",
+    "$$\\langle \\psi \\vert  e^{i \\hat{H} m \\delta t}   \\hat{H} e^{-i \\hat{H} n \\delta t} \\vert \\psi \\rangle$$\n",
+    "$$\\langle \\psi \\vert \\hat{H} e^{-i \\hat{H} (n-m) \\delta t} \\vert \\psi \\rangle$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports and definitions\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "import numpy as np\n",
+    "import scipy as sp\n",
+    "import matplotlib.pylab as plt\n",
+    "import warnings\n",
+    "warnings.filterwarnings('ignore')\n",
+    "\n",
+    "from qiskit.quantum_info import SparsePauliOp, Operator\n",
+    "from qiskit.circuit import Parameter\n",
+    "from qiskit import QuantumCircuit, QuantumRegister, transpile\n",
+    "from qiskit.circuit.library import PauliEvolutionGate\n",
+    "from qiskit.synthesis import SuzukiTrotter, MatrixExponential\n",
+    "from qiskit.providers.fake_provider import Fake20QV1\n",
+    "from qiskit.primitives import Estimator\n",
+    "\n",
+    "def solve_regularized_gen_eig(h, s, threshold, k=1, return_dimn=False):\n",
+    "    s_vals, s_vecs = sp.linalg.eigh(s)\n",
+    "    s_vecs = s_vecs.T\n",
+    "    good_vecs = np.array([vec for val, vec in zip(s_vals, s_vecs) if val > threshold])\n",
+    "    h_reg = good_vecs.conj() @ h @ good_vecs.T\n",
+    "    s_reg = good_vecs.conj() @ s @ good_vecs.T\n",
+    "    if k==1:\n",
+    "        if return_dimn:\n",
+    "            return sp.linalg.eigh(h_reg, s_reg)[0][0], len(good_vecs)\n",
+    "        else:\n",
+    "            return sp.linalg.eigh(h_reg, s_reg)[0][0]\n",
+    "    else:\n",
+    "        if return_dimn:\n",
+    "            return sp.linalg.eigh(h_reg, s_reg)[0][:k], len(good_vecs)\n",
+    "        else:\n",
+    "            return sp.linalg.eigh(h_reg, s_reg)[0][:k]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Hamiltonian\n",
+    "Let's consider the Heisenberg Hamiltonian for $N$ qubits on a linear chain: $H=-J \\sum_{i,j}^N Z_i Z_j + J \\sum_{i,j}^N X_i X_j + Y_i Y_j$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "SparsePauliOp(['ZZIIIIIIII', 'IZZIIIIIII', 'IIZZIIIIII', 'IIIZZIIIII', 'IIIIZZIIII', 'IIIIIZZIII', 'IIIIIIZZII', 'IIIIIIIZZI', 'IIIIIIIIZZ', 'XXIIIIIIII', 'IXXIIIIIII', 'IIXXIIIIII', 'IIIXXIIIII', 'IIIIXXIIII', 'IIIIIXXIII', 'IIIIIIXXII', 'IIIIIIIXXI', 'IIIIIIIIXX', 'YYIIIIIIII', 'IYYIIIIIII', 'IIYYIIIIII', 'IIIYYIIIII', 'IIIIYYIIII', 'IIIIIYYIII', 'IIIIIIYYII', 'IIIIIIIYYI', 'IIIIIIIIYY'],\n",
```

Comment:
[Suggestion only] I find the complex representation of the coefficients to get in the way of understanding things. I would either print H_op with a conversion to ints or print H_tot which gives the coefficients in integer form next to each operator (which is also nice)